core/drivers/crypto/caam/utils/utils_mem.c

```diff
@@ -0,0 +1,539 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/**
+ * @copyright 2018-2019 NXP
+ *
+ * @file    utils_mem.c
+ *
+ * @brief   Memory management utilities.\n
+ *          Primitive to allocate, free memory.
+ */
+
+/* Global includes */
+#include <arm.h>
+#include <malloc.h>
+#include <mm/core_memprot.h>
+#include <mm/tee_mmu.h>
+#include <string.h>
+
+/* Local includes */
+#include <caam_common.h>
+
+/* Utils includes */
+#include <caam_utils_mem.h>
+
+/**
+ * @brief   Definition of some armv7 Cache Size Register fields
+ */
+#define CCSIDR_LINESIZE_SHIFT	0
+#define CCSIDR_LINESIZE_MASK	0x7
+
+/**
+ * @brief   CAAM Descriptor address alignment
+ */
+#ifdef ARM64
+#define DESC_START_ALIGN	(64 / 8)
+#else
+#define DESC_START_ALIGN	(32 / 8)
+#endif
+
+/**
+ * @brief   Check if pointer \a p is aligned with \a align
+ */
+#define IS_PTR_ALIGN(p, align)	(((uintptr_t)(p) & (align - 1)) == 0)
+
+/**
+ * @brief   Check if size \a size is aligned with \a align
+ */
+#define IS_SIZE_ALIGN(size, align)	(ROUNDUP(size, align) == size)
+
+/**
+ * @brief   Align the pointer \a p to be align with \a align modulus
+ */
+#define PTR_ALIGN(p, align)	(p + ((align - (p % align)) % align))
+
+/**
+ * @brief   Cache line size in bytes
+ */
+static uint16_t cacheline_size;
+
+/**
+ * @brief   Read the system cache line size.\n
+ *          Get the value from the ARM system configration register
+ */
+static void read_cacheline_size(void)
+{
+	uint32_t value;
+
+#ifdef ARM64
+	asm volatile ("mrs %0, ctr_el0" : "=r" (value));
+	cacheline_size = 4 << ((value >> CTR_DMINLINE_SHIFT)
+						& CTR_DMINLINE_MASK);
+#else
+	asm volatile ("mrc p15, 1, %0, c0, c0, 0" : "=r" (value));
```

Comment:
This assumes that the correct cache has been selected already. Why not use `read_ctr()` instead?